### PR TITLE
refactor: batchKey logic

### PR DESCRIPTION
**Pull request by karatakis**
I rewrote the documentation because it was unclear how to use the API based on feedback from here https://github.com/tailcallhq/tailcall/pull/2495

### CODE REVIEW

File: docs/directives.md [hunk 1]
```diff
@@ -882,7 +882,7 @@ type Mutation {
 
 ### query
 
-Represents the API call's query parameters, either as a static object or with dynamic parameters using Mustache templates. These parameters append to the URL.
+Represents the API call's query parameters, either as a static object or with dynamic parameters using Mustache templates. These parameters append to the URL. The `key` parameter of `query` is optional. If it is not present TailCall uses the last item from `batchKey` parameter of `@http` directive to derive the `key` parameter, so remember to adjust the order of the items accordingly.
```
Comment (tusharmath) [inline docs/directives.md]: I think this note should be written under an `:::important` section of `batchKey`

**Comment by tusharmath** [comment]
@karatakis Can you please resolve the conflicts here.

File: docs/directives.md (hunk 1 above)
Comment (tusharmath) [reply docs/directives.md]: Let's provide an actual example here for more clarity.

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by karatakis** [comment]
@tusharmath I rewrote the docs to describe how Tailcall behaves when batchKey and query are present. I also changed the batchKey example to be more generic and describe better the edge case.

**Review by tusharmath — APPROVED** [review]

**Comment by tusharmath** [comment]
Thanks!

**Pull request merged**